src/main.rs

```diff
@@ -28,19 +37,45 @@ struct CompilerOptions {
     /// Set the maximum stack size for the current running instance.
     #[clap(short, long, default_value = "10000")]
     stack_size: u32,
+
+    /// Run the compiler in debug mode
+    #[clap(short, long)]
+    debug: bool,
 }
 
 fn main() {
     let opts: CompilerOptions = CompilerOptions::parse();
 
-    println!("Stack_size is {}", opts.stack_size);
+    // print recieved cmdargs, if debug is specified
+    if opts.debug {
+        println!("Stack_size is {}", opts.stack_size);
 
-    for path in opts.includes.into_iter() {
-        println!("Running with {}", path);
+        for path in opts.includes.into_iter() {
```

Comment:
Avoid using `into_iter()` when iterating over something because it will copy each element every time you loop through it. Use `iter()` instead to get references to the elements.

Reply:
Nice, ill take note for the future.